packages/core/core/src/TransformerRunner.js

```diff
@@ -337,29 +341,37 @@ async function summarizeRequest(
     // lazily read it at a later point.
     content = Buffer.from([]);
     size = 0;
-    hash = await md5FromReadableStream(
-      createReadStream(req.filePath).pipe(
-        new TapStream(buf => {
-          size += buf.length;
-          if (content instanceof Buffer) {
-            if (size > BUFFER_LIMIT) {
-              // if buffering this content would put this over BUFFER_LIMIT, replace
-              // it with a stream
-              content = createReadStream(req.filePath);
-            } else {
-              content = Buffer.concat([content, buf]);
+    [hash, fileMode] = await Promise.all([
+      md5FromReadableStream(
+        createReadStream(req.filePath).pipe(
+          new TapStream(buf => {
+            size += buf.length;
+            if (content instanceof Buffer) {
+              if (size > BUFFER_LIMIT) {
+                // if buffering this content would put this over BUFFER_LIMIT, replace
+                // it with a stream
+                content = createReadStream(req.filePath);
+              } else {
+                content = Buffer.concat([content, buf]);
+              }
             }
-          }
-        })
-      )
-    );
+          })
+        )
+      ),
+      // `mode` is a decimal representation of the more common octal format of
+      // unix file permissions. Assign it to the asset so it can be used to determine
+      // the permissions of the output bundle. Node's documentation on this is pretty
+      // sparse as it's probably just the result of a `stat` syscall. More info:
+      // https://stackoverflow.com/questions/11775884/nodejs-file-permissions
+      stat(req.filePath).then(stats => stats.mode)
```

Comment:
hmm an extra stat call for every asset won't be cheap. wonder if we could just stat the entry file in the packager runner rather than propagating `fileMode` everywhere...

Reply:
Awesome idea. Just implemented it 😄